### PR TITLE
Progress reporting for downloading mdb

### PR DESCRIPTION
**Pull request by seanses**
Reports progress when downloading MerkleDB on repo cloning. This can take a while on large repos.
Screenshot 2024-01-12 at 5 38 00 PM

Fix https://github.com/xetdata/xethub/issues/4939

### CODE REVIEW

**Review by znation — APPROVED** [review]

**Comment by allisonwhilden** [comment]
Always awesome to have status feedback for users. Thank you @seanses!

I have a slight question as to if we can do more to improve clarity to users in the wording. Questions I was asking myself:
- Do users understand what the MerkleDB is? Does that matter?
- Is there a difference between **Receiving** objects and **Retrieving** MerkleDB? 
- Is it common to put a version number before the colon? ("Git-Xet 0.13.0:" vs. "Git-Xet:")

These are all very minor thoughts and by no means are any reason to not merge this. Happy to just be something I can learn later. In part I'm just trying to understand some details to how we approach our CLI UX (and what are the opportunities for improvement). Thanks!

**Review by allisonwhilden — APPROVED** [review]

**Comment by seanses** [comment]
@allisonwhilden Good questions! 
- MerkleDB may be a term too technical, how about "metadata"? In https://github.com/xetdata/xet-core/pull/230 when the actual file content is downloaded Git-Xet prints "Git-Xet 0.12.8: Downloading data", I wanted to pick a different term to show they are two steps.
- I think "Receiving objects" comes from git, that relates to downloading everything under .git/objects, which is essentially contains basic git repo information.
- A version number helps us debug when a log file is not provided, especially at the moment a client can change a lot between versions. I lean toward keeping it if not too wordy.

**Comment by allisonwhilden** [comment]
Thank you! I like metadata better. Your other comments make sense to me. Thanks!
